### PR TITLE
 Finalizing Tests

### PR DESCRIPTION
**Pull request by TcgVanguardTroll**
These Are the tests for the additional scenario placed within Week 5, they include a

- Including  an optional attendee C who has an all-day event. The same three time slots should be returned as when C was not invited.

- Based on `everyAttendeeIsConsidered`, adds an optional attendee C who has an event between 8:30 and 9:00. Now only the early and late parts of the day should be returned.

- Based on the `justEnoughRoom` method , adds an optional attendee B who has an event between 8:30 and 8:45. The optional attendee should be ignored since considering their schedule would result in a time slot smaller than the requested time.

- No mandatory attendees, just two optional attendees with several gaps in their schedules. Those gaps should be identified and returned.

- No mandatory attendees, just two optional attendees with no gaps in their schedules. `query` should return that no time is available.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 import java.util.List;
 import org.junit.Assert;
 import org.junit.Before;
-import org.junit.Test;
+import org.junit.Test
```
Comment (tcwanggoogle) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java]: Removed by accident?

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 2]
```diff
@@ -82,14 +82,14 @@ public void noOptionsForTooLongOfARequest() {
   public void eventSplitsRestriction() {
     // The event should split the day into two options (before and after the event).
     Collection<Event> events = Arrays.asList(new Event("Event 1",
-        TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES), Arrays.asList(PERSON_A)));
+            TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES), Arrays.asList(PERSON_A)));
```
Comment (tcwanggoogle) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java]: Are the spaces added by formatter, or by accident? Same with below

**Review by tcwanggoogle — COMMENTED** [review]
Are the file/function which these tests cover already checked in? (I am just trying to figure out the sequence of the split PRs) In practice, we either check in tests together with the functionality we added, or after.

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java (hunk 1 above)
Comment (TcgVanguardTroll) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java]: This was removed by accident and is noted. My thought process was to work on tests that I'm sure will work and add them in a pull request then once I finalize my implementation they should pass.

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java (hunk 2 above)
Comment (TcgVanguardTroll) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java]: Forgot to format

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java (hunk 1 above)
Comment (kmh287) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java]: Adding the semicolon back should eliminate the diff for this file

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 3]
```diff
@@ -104,19 +104,80 @@ public void everyAttendeeIsConsidered() {
...
     Collection<TimeRange> actual = query.query(events, request);
     Collection<TimeRange> expected =
-        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
-            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
-            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+            Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+                    TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+                    TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
```
Comment (kmh287) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java]: I strongly recommend pulling in the [Truth](www.truth.dev) library to simplify these assertions. Then you can write

assertThat(query.query(events, request)).containsExactly(....)

**Comment by kmh287** [comment]
+1 on Tiancong's comment. TDD absolutely does call for writing the tests before the implementation, but both end up together in the same PR. Continuous integration systems including those we use internally at Google prevent anyone form submitting code if it causes tests to fail. I recommend adding the implementation in this PR as well.

**Review by kmh287 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**